core/src/main/java/smashdudes/gameplay/state/playerstate/air/JumpState.java

```diff
@@ -25,11 +23,28 @@ public void innerUpdate(float dt)
             float speed = v.airSpeed;
             if(ci.currentState.left)
             {
-                v.velocity.x -= speed;
+                v.velocity.x = -speed;
             }
             if(ci.currentState.right)
             {
-                v.velocity.x += speed;
+                v.velocity.x = speed;
+            }
+        }
+
+        if(!ci.currentState.up)
+        {
+            if(v.velocity.y > 0.1f)
+            {
+                // TODO (danielg): We may want this This jump-deceleration (extra gravity) to be set per character
+                float decelerationThisFrame = 128 * dt;
+                if (Math.abs(v.velocity.y) > decelerationThisFrame)
+                {
+                    v.velocity.y -= Math.signum(v.velocity.y) * decelerationThisFrame;
+                }
+                else
+                {
+                    v.velocity.y = 0;
+                }
```

Comment:
This mirrors the deceleration in the MovementSystem.
Essentially we apply extra gravity (deceleration towards 0) after we release the jump key. This causes us to do a short jump, but still have some upward momentum after we release the key.